### PR TITLE
{chem}[iomkl/2017b] QuantumEspresso v6.2.1

### PR DESCRIPTION
**Pull request by akesandgren**
(created using `eb --new-pr`)
New version of QuantumEspresso (6.2.1). Requires easyblock PR https://github.com/easybuilders/easybuild-easyblocks/pull/1312

### CODE REVIEW

File: easybuild/easyconfigs/q/QuantumESPRESSO/QuantumESPRESSO-6.2.1-iomkl-2017b.eb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+checksums = [
+    '6cdd4c8883b4a2262ac98ea452f845ed935294b3988a84b238d833558eb90f0c',  # qe-6.2.1.tar.gz
+    'ee90108d4bc4aa6a1cf16d72abebcb3087cf6c1007d22dda269eb7e7076bddca',  # wannier90-2.1.0.tar.gz
+    # QuantumESPRESSO-6.2.1_fix_missing_include_dir_in_COUPLE.patch
+    '44751fe9d1e6eede4ba331e7216e222960ff19ba9170aaa0cd4393924aa54fbc',
+]
+
+dependencies = [
+    ('HDF5', '1.10.1'),
+    ('FoX', '4.1.2'),
```
Comment (migueldiascosta) [inline easybuild/easyconfigs/q/QuantumESPRESSO/QuantumESPRESSO-6.2.1-iomkl-2017b.eb]: the easyblock now prevents `FoX` from being used as external module

File: easybuild/easyconfigs/h/HDF5/HDF5-1.10.1-iomkl-2017b.eb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name = 'HDF5'
+version = '1.10.1'
+
+homepage = 'https://support.hdfgroup.org/HDF5/'
```
Comment (boegel) [inline easybuild/easyconfigs/h/HDF5/HDF5-1.10.1-iomkl-2017b.eb]: should be updated to https://portal.hdfgroup.org/display/support (see #6687)

File: easybuild/easyconfigs/h/HDF5/HDF5-1.10.1-iomkl-2017b.eb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+toolchain = {'name': 'iomkl', 'version': '2017b'}
+toolchainopts = {'pic': True, 'usempi': True}
+
+source_urls = ['https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-%(version_major_minor)s/hdf5-%(version)s/src']
+sources = [SOURCELOWER_TAR_GZ]
+checksums = ['048a9d149fb99aaa1680a712963f5a78e9c43b588d0e79d55e06760ec377c172']
+
+dependencies = [
+    ('zlib', '1.2.11'),
+    ('Szip', '2.1.1'),
```
Comment (boegel) [inline easybuild/easyconfigs/h/HDF5/HDF5-1.10.1-iomkl-2017b.eb]: missing `libxml2`, see #6498

File: easybuild/easyconfigs/q/QuantumESPRESSO/QuantumESPRESSO-6.2.1-iomkl-2017b.eb [hunk 4]
```diff
@@ -0,0 +1,47 @@
+name = 'QuantumESPRESSO'
+version = '6.2.1'
+
+homepage = 'http://www.pwscf.org/'
```
Comment (boegel) [inline easybuild/easyconfigs/q/QuantumESPRESSO/QuantumESPRESSO-6.2.1-iomkl-2017b.eb]: `https://www.quantum-espresso.org/`?

**Comment by pescobar** [comment]
Test report by @pescobar
**FAILED**
Build succeeded for 0 out of 3 (3 easyconfigs in this PR)
login20.cluster.bc2.ch - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 v2 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/9b12ef779f990d1350dab94a5d1321bf for a full test report.

**Comment by boegel** [comment]
@pescobar forgot `--robot`?

**Comment by boegel** [comment]
@akesandgren Does this work for you?

Fails for me with
```
read_upf_schema.f90(18): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [FOX_DOM]
        USE FoX_Dom
------------^
```

(see upcoming test report)

**Comment by migueldiascosta** [comment]
@akesandgren this version needs something like

```diff
--- qe-6.2.1/install/extlibs_makefile.orig	2019-05-09 09:41:29.000000000 +0800
+++ qe-6.2.1/install/extlibs_makefile	2019-05-09 09:42:01.000000000 +0800
@@ -80,7 +80,7 @@
 	if test ! -d ../FoX; then \
         mkdir ../FoX; \
 	(gzip -dc ../archive/fox.tgz | (cd ../FoX; tar -xvf -)); \
-	cd ../FoX/fox/; export FC=$(F90); ./configure --prefix=$(TOPDIR)/FoX ;\
+	cd ../FoX/fox/; export FC=$(F90); export FCFLAGS="$(FOX_FLAGS)"; \
+	unset LIBS; ./configure --prefix=$(TOPDIR)/FoX ;\
 	$(MAKE) install; cd ../; rm -fr fox;fi
 # ELPA
 libelpa : libelpa_$(ELPA_LIBS_SWITCH)
```

(this was fixed in `6.3`)

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/8e3cf3354505062bc2aa7ba62b35b668 for a full test report.

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
grc-cluster1 - Linux centos 6.10, Intel(R) Xeon(R) CPU E5-2640 0 @ 2.50GHz, Python 2.7.14
See https://gist.github.com/7afff4a579a4e99505d0a7d34967399b for a full test report.

**Comment by migueldiascosta** [comment]
@akesandgren I suppose it doesn't hurt but the `FoX` easyconfig is still here and no longer necessary, do you want to remove it?

**Comment by akesandgren** [comment]
Yeah, I just forgot... removing

**Comment by migueldiascosta** [comment]
Going in, thanks @akesandgren!

**Pull request merged**